pkg/cassandra/config/schema.go

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package config
+
+import (
+	"bytes"
+	"embed"
+	"errors"
+	"fmt"
+	"text/template"
+	"time"
+
+	"github.com/jaegertracing/jaeger/pkg/cassandra"
+)
+
+//go:embed v004-go-tmpl.cql.tmpl
+var schemaFile embed.FS
+
+type TemplateParams struct {
+	// Keyspace in which tables and types will be created for storage
+	Keyspace string
+	// Replication is the replication strategy used. Ex: "{'class': 'NetworkTopologyStrategy', 'replication_factor': '1' }"
+	Replication string
+	// CompactionWindowInMinutes is constructed from CompactionWindow for using in template
+	CompactionWindowInMinutes int64
+	// TraceTTLInSeconds is constructed from TraceTTL for using in template
+	TraceTTLInSeconds int64
+	// DependenciesTTLInSeconds is constructed from DependenciesTTL for using in template
+	DependenciesTTLInSeconds int64
+}
+
+func constructTemplateParams(cfg Schema) TemplateParams {
+	return TemplateParams{
+		Keyspace:                  cfg.Keyspace,
+		Replication:               fmt.Sprintf("{'class': 'NetworkTopologyStrategy', 'replication_factor': '%v' }", cfg.ReplicationFactor),
+		CompactionWindowInMinutes: int64(cfg.CompactionWindow / time.Minute),
+		TraceTTLInSeconds:         int64(cfg.TraceTTL / time.Second),
+		DependenciesTTLInSeconds:  int64(cfg.DependenciesTTL / time.Second),
+	}
+}
+
+func getQueryFileAsBytes(fileName string, params TemplateParams) ([]byte, error) {
+	tmpl, err := template.ParseFS(schemaFile, fileName)
+	if err != nil {
+		return nil, err
+	}
+
+	var result bytes.Buffer
+	err = tmpl.Execute(&result, params)
+	if err != nil {
+		return nil, err
+	}
+
+	return result.Bytes(), nil
+}
+
+func getQueriesFromBytes(queryFile []byte) ([]string, error) {
+	lines := bytes.Split(queryFile, []byte("\n"))
+
+	var extractedLines [][]byte
+
+	for _, line := range lines {
+		// Remove any comments, if at the end of the line
+		commentIndex := bytes.Index(line, []byte(`--`))
+		if commentIndex != -1 {
+			// remove everything after comment
+			line = line[0:commentIndex]
+		}
+
+		if len(line) == 0 {
```

Comment:
it would make sense to trim spaces before checking for len=0.

Reply:
Done. Thanks